core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/report/detail/ReportDetailViewModel.kt

```diff
@@ -45,53 +61,106 @@ class ReportDetailViewModel(
+                        icon = FabUiState.FabIcon.EDIT,
+                        onClick = this@ReportDetailViewModel::onClickEdit
+                    ),
+                title = "Graph",
             )
         }
+        viewModelScope.launch {
+            _uiState.whenSubscribed {
+                launchIfHasPermission(
+                    setLoadingState = true,
+                    permissionCheck = { true }
+                ) {
+                    val siteFlow = activeRepo.reportDao().findByUidLive(reportUid)
+                    launch {
+                        siteFlow.collect { report ->
+                            try {
+                                val reportNonNull = report
+                                    ?: throw IllegalStateException("Report not found for uid $reportUid")
+                                val optionsJson = reportNonNull.reportOptions
+                                val reportTitle = reportNonNull.reportTitle
+
+                                val parsedOptions = when {
+                                    !optionsJson.isNullOrBlank() -> {
+                                        try {
+                                            Json.decodeFromString(
+                                                ReportOptions2.serializer(),
+                                                optionsJson
+                                            )
+                                        } catch (e: Exception) {
+                                            throw IllegalArgumentException("Invalid report options format")
+                                        }
+                                    }
+
+                                    !reportTitle.isNullOrBlank() -> {
+                                        ReportOptions2(title = reportTitle)
+                                    }
 
-        launchIfHasPermission(
-            setLoadingState = true,
-            permissionCheck = { true }
-        ) {
-            launch {
-                navResultReturner.filteredResultFlowForKey(RESULT_KEY_REPORT_DETAIL)
-                    .collect { result ->
-                        val report = result.result as? Report ?: return@collect
-                        getReport(report)
+                                    else -> throw IllegalStateException("Report $reportUid has no options or title")
+                                }
+
+                                _uiState.update { it.copy(reportOptions2 = parsedOptions) }
+
+                                val request = RunReportUseCase.RunReportRequest(
+                                    reportUid = reportUid,
+                                    reportOptions = parsedOptions,
+                                    accountPersonUid = activeUserPersonUid,
+                                    timeZone = TimeZone.currentSystemDefault()
+                                )
+                                runReport(request)
```

Comment:
This needs to change to something like:


```
runReportUseCase(request).collect { reportResult -> 
                                    _uiState.update { prev ->
                                        prev.copy(reportResults = reportResult.results)
                                    }
                                }
```
Add try/catch - add somewhere in the UI to show the error message.